### PR TITLE
Create store

### PR DESCRIPTION
**Pull request by RonaTross**
@yonbergman @maximveksler cr away 📝 
Yes I got a little carried away so the branch is called create_store but contains a bit more that 😅

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import {Provider} from 'react-redux'
+import {UpcomingTalkGroup, PastTalkGroup} from './TalkGroup'
+import talksStore from './talksStore'
+import {Navbar} from './Navbar'
+
+/**
+* The entire app.
+*/
```
Comment (yonbergman) [inline App.js]: remove this comments

File: App.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class App extends React.Component {
+
+    constructor(props) {
+       super(props);
+       this.displayName = 'App';
+   }
+
+    render() {
+        console.log(Navbar)
```
Comment (yonbergman) [inline App.js]: remove the console log

File: App.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        return  <Provider store={talksStore}>
```
Comment (yonbergman) [inline App.js]: I prefer we start using the following syntax for multi line renders

```
render() {
  return (
    <div>
      <moredivs/>
    </div>
  );
}
```

File: App.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+                    <div>
+                        <Navbar />
```
Comment (yonbergman) [inline App.js]: extra space

File: DraggableTalk.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import {Panel, Button} from 'react-bootstrap'
+import {sortable} from 'react-anything-sortable';
+import TalkModel from './talkModel'
+import Talk from './Talk'
+import {deleteTalk} from './updateServerUtils'
+
+/**
+ * Displays a talk that can be dragged up or down (to reorder the list of talks).
+ */
```
Comment (yonbergman) [inline DraggableTalk.js]: I'm really against comments that are unnecessary. Draggable talk is a great name

File: DraggableTalk.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+@sortable
+class DraggableTalk extends React.Component {
+    
+    constructor(props) {
+        super(props);
+        this.displayName = 'DraggableTalk';
+    }
+
+    deleteTalk() {
+        deleteTalk(this.props.talk.id, this.props.deleteTalkFromStore)
```
Comment (yonbergman) [inline DraggableTalk.js]: I would set the name from the store to something else to differentiate these two calls

File: NewTalkForm.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import {connect} from 'react-redux'
+import {Button, FormGroup, FormControl, ControlLabel} from 'react-bootstrap'
+import {addTalk, deleteTalk} from './updateServerUtils'
+
+/**
+ * A form to fill in order to add a new talk.
+ */
```
Comment (yonbergman) [inline NewTalkForm.js]: comments

File: NewTalkForm.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+ 
+        var validUserOptions = userNames.map((userName) => {
+            return <option value={userName} key={userName}>{userName}</option>})
+        var placeHolderOption = <option disabled key='placeholder' value='pickUser'>Who is giving the talk</option>
+
+        return     <form onSubmit={this.submit.bind(this)}>
+                      <FormGroup controlId='user'>
+                          <FormControl componentClass='select' defaultValue='pickUser'>
+                            {[placeHolderOption, validUserOptions]}
+                        </FormControl>
```
Comment (yonbergman) [inline NewTalkForm.js]: nice

File: TalkGroup.js [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+    handleSort(data) {
+        reorderTalks(data, this.props.reorderTalksInStore)
+    }
+
+    generateTalkComponents() {
+        var talks = this.props.talks
+        var talksOrder = this.props.talksOrder
+        
+        //TODO admin state
+        if(this.props.draggableTalks) {
```
Comment (yonbergman) [inline TalkGroup.js]: spacing is off

File: talkModel.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+/**
+ * Contains basic info about a talk.
+ */
```
Comment (yonbergman) [inline talkModel.js]: comment

File: talksStore.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+var initialState = {
+    upcomingTalks: {1: new TalkModel(1, 'Very interesting topic', 'Josh', 'This is some very intersting talk'), 2: new TalkModel(2, 'Another very interesting topic', 'Josh Junior', 'Bla bla bla bla bla')},
+    upcomingTalksOrder: [1,2],
+    pastTalks: {3: new TalkModel(3, 'So last year', 'Mike', 'איך זה עם עברית?'), 4: new TalkModel(4, '2000 an\' late', 'Hike', 'ככה ככה')},
+    pastTalksOrder: [3,4]
+}
+var talksStore = createStore(talksRedeucer, initialState)
+
```
Comment (yonbergman) [inline talksStore.js]: nice 👍

**Comment by yonbergman** [comment]
Store stuff looks great, a few minor comments elsewhere

File: TalkGroup.js (hunk 9 above)
Comment (RonaTross) [inline TalkGroup.js]: too many blank lines?

**Pull request merged**